examples/rules.mk

```diff
@@ -200,27 +238,21 @@ styleclean: $(STYLECHECKFILES:=.styleclean)
 
 ifeq ($(STLINK_PORT),)
 ifeq ($(BMP_PORT),)
-ifeq ($(OOCD_SERIAL),)
-%.flash: %.hex
+ifeq ($(OOCD_FILE),)
+%.flash: %.elf
```

Comment:
This is only for the "make flash" target, which is using openocd.  WIthout this change, the hex file would have been generated just for flashing, which openocd doesn't need.   The examples repo has only made .elf files by default since 
3af1f8d43d481b4d904ca85fdfb96c7bbee3705b in Feb 2014.  However, "make bin" or "make hex" in either the root, or in an individual example directory will build those files on demand.